src/DIRAC/Resources/IdProvider/Utilities.py

```diff
@@ -0,0 +1,87 @@
+""" Utilities for the IdProvider package
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+__RCSID__ = "$Id$"
+
+from DIRAC import S_OK, S_ERROR, gConfig
+
+
+def getSettingsNamesForIdPIssuer(issuer):
+  """ Get identity providers for issuer
+
+      :param str issuer: issuer
+
+      :return: S_OK(list)/S_ERROR()
+  """
+  names = []
+  result = getProvidersForInstance('Id')
+  if not result['OK']:
+    return result
+  for name in result['Value']:
+    nameIssuer = gConfig.getValue('/Resources/IdProviders/%s/issuer' % name)
+    if nameIssuer and issuer.strip('/') == nameIssuer.strip('/'):
+      names.append(name)
+  return S_OK(names) if names else S_ERROR('Not found provider with %s issuer.' % issuer)
+
+
+def getSettingsNamesForClientID(clientID):
+  """ Get identity providers for clientID
+
+      :param str clientID: clientID
+
+      :return: S_OK(list)/S_ERROR()
+  """
+  names = []
+  result = getProvidersForInstance('Id')
+  if not result['OK']:
+    return result
+  for name in result['Value']:
+    res = gConfig.getValue('/Resources/IdProviders/%s/client_id' % name)
+    if res and clientID == res:
+      names.append(name)
+  return S_OK(names) if names else S_ERROR('Not found provider with %s clientID.' % clientID)
+
+
+def getProvidersForInstance(instance, providerType=None):
+  """ Get providers for instance
+
+      :param str instance: instance of what this providers
```

Comment:
Apart from "Id", what could `instance` be?

Reply:
There are also proxy providers